app/model/ReindexProgress.scala

```diff
@@ -0,0 +1,85 @@
+package model;
+
+import com.amazonaws.services.dynamodbv2.document.Item
+import org.cvogt.play.json.Jsonx
+import play.api.libs.json._
+import play.api.Logger
+import scala.util.control.NonFatal
+
+case class ReindexProgress(`type`: String, status: String, docsSent: Int, docsExpected: Int) {
+  def toItem() = Item.fromJSON(Json.toJson(this).toString())
+  def toCapiForm() = CapiReindexProgress(status, docsSent, docsExpected)
+}
+
+object ReindexProgress {
+  val TagTypeName = "tag"
+  val SectionTypeName = "section"
+
+  val InProgress = "in progress"
+  val Failed = "failed"
+  val Completed = "completed"
+  val Cancelled = "cancelled"
+  val Unknown = "unknown"
```

Comment:
It's only used when the dynamo table we're storing our state in is empty for some reason. Either someone cleared it out or there has never been a reindex ran from that table.